### ISSUE TITLE
Fold code block

### ISSUE THREAD

**Issue by BindingOx** [mProjectsCode/obsidian-shiki-plugin#42]
**Is your feature request related to a problem? Please describe.**
Allow folding codeblocks to save space

**Describe the solution you'd like**
Fold code
- If the header is displayed (either by defining filename or other way explained below), you can click on the header to fold the code block below it

**Additional context**
CodeblockCustomizer has this feature https://github.com/mugiwara85/CodeblockCustomizer#folding

**Comment by simonfossom** [comment]
only this plugin needs to be added: 

[Collapsible Sections | Expressive Code](https://expressive-code.com/plugins/collapsible-sections/)

**Comment by mProjectsCode** [comment]
> only this plugin needs to be added:
> 
> [Collapsible Sections | Expressive Code](https://expressive-code.com/plugins/collapsible-sections/)

That plugin is already included.

**Comment by sergeevabc** [comment]
Sorry, but I still don't understand how to collapse the code block.
You can ```language collapse={1-5}``` to collapse a few lines, but how to fold all the block.

**Comment by simonfossom** [comment]
> Sorry, but I still don't understand how to collapse the code block. You can `language collapse={1-5}` to collapse a few lines, but how to fold all the block?

Yeah... you cannot. I have a expantion collapse={1-1000} and that's the best you can do. 
@mProjectsCode didn't implement folding the whole block to a single line to work like callout. 

best alternative is use expanding templater templalte with admonitions
`````
````ad-code
title: Some title
collapse: true
  ```js 
  // code here
  ```
````
`````
it's ugly in source, it works in preview. But that's it if you want to use it and cannot extend it yourself. 🤷‍♂️
